app/models/manageiq/providers/ibm_power_hmc/infra_manager/event_target_parser.rb

```diff
@@ -20,35 +20,35 @@ def parse
     case ems_event.event_type
     when "MODIFY_URI", "ADD_URI", "DELETE_URI" # Damien: INVALID_URI?
       $ibm_power_hmc_log.info("#{self.class}##{__method__} #{ems_event.event_type} #{ems_event.full_data}")
-      elems = elem(raw_event[:data])
-      case elems[:type]
+      elems = raw_event[:data].match(/^.+\/rest\/api\/uom\/(ManagedSystem\/(?<manager_uuid>.+)\/)?(?<type>.+)\/(?<uuid>.+)$/).named_captures
+      case elems["type"]
       when "ManagedSystem"
-        new_targets << {:assoc => :hosts, :ems_ref => elems[:uuid]}
+        new_targets << {:assoc => :hosts, :ems_ref => elems["uuid"]}
       when "LogicalPartition", "VirtualIOServer"
         # raw_event[:detail] contains information about the properties that
         # have changed (e.g. RMCState, PartitionName, PartitionState etc...)
         # This may be used to perform quick property REST API calls to the HMC
         # instead of querying the full LPAR data.
-        if elems[:uuid] == NO_UUID_VALUE
-          $ibm_power_hmc_log.info("#{self.class}##{__method__} #{elems[:type]} Missing LPAR UUID.  Escalating to full refresh for EMS: [#{ems.name}], id: [#{ems.id}].")
+        if elems["uuid"] == NO_UUID_VALUE
+          $ibm_power_hmc_log.info("#{self.class}##{__method__} #{elems["type"]} Missing LPAR UUID.  Escalating to full refresh for EMS: [#{ems.name}], id: [#{ems.id}].")
           target_collection << ems
         else
-          new_targets << {:assoc => :vms, :ems_ref => elems[:uuid]}
+          new_targets << {:assoc => :vms, :ems_ref => elems["uuid"]}
         end
       when "VirtualSwitch", "VirtualNetwork"
-        if elems.key?(:manager_uuid)
-          new_targets << {:assoc => :hosts, :ems_ref => elems[:manager_uuid]}
+        if elems["manager_uuid"]
```

Comment:
This changed from .key? to truthy because if it isn't matched the payload will have `{"manager_uuid" => nil}`